yarn-project/circuits.js/src/structs/revert_code.ts

```diff
@@ -16,28 +16,47 @@ function isRevertCodeEnum(value: number): value is RevertCodeEnum {
  * Wrapper class over a field to safely represent a revert code.
  */
 export class RevertCode {
-  private code: Fr;
+  private code: number;
   private constructor(e: RevertCodeEnum) {
-    this.code = new Fr(e);
+    this.code = e.valueOf();
   }
-
   static readonly OK: RevertCode = new RevertCode(RevertCodeEnum.OK);
   static readonly REVERTED: RevertCode = new RevertCode(RevertCodeEnum.REVERTED);
 
   public equals(other: RevertCode): boolean {
-    return this.code.equals(other.code);
+    return this.code == other.code;
```

Comment:
This could be a strict equal (even through they're both numbers)
```suggestion
    return this.code === other.code;
```